var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -14,6 +14,8 @@ class Hypre(Package):
 
     # hypre does not know how to build shared libraries on Darwin
     variant('shared', default=sys.platform!='darwin', description="Build shared library version (disables static library)")
+    # SuperluDist have conflicting headers with those in Hypre
+    variant('internal-superlu', default=True, description="Use internal Superlu routines")
```

Comment:
i think you mixed a bit packages. If I understand you correctly, here, in `hypre` we do `+superlu` (build itnernal superlu). It does **not** (AFAIK) build against external one (thus possible collision in headers). Now another package, `superlu_dist` (!= superlu), also has those headers from `superlu` (serial version). Finally, `petsc` can be build against both `hypre` (preconditioner) and `superlu_dist` (direct parallel solver).